From f531dad71f45778fd270e0fbb3aa36e552362c7d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 25 Sep 2025 00:34:41 +0200
Subject: [PATCH] Use weasyprint instead of typst for PDF export

Typst has some limitations for referenced files, only relative path are
allowed inside the working directory. The weasyprint engine has no such
limitation, but sill provide a lightweight way to export to PDF.
---
 apostrophe/export_dialog.py | 11 ++++-------
 apostrophe/main_window.py   |  2 --
 data/media/formats.json     |  2 +-
 3 files changed, 5 insertions(+), 10 deletions(-)

diff --git a/apostrophe/export_dialog.py b/apostrophe/export_dialog.py
index f3ab16b..c24652c 100644
--- a/apostrophe/export_dialog.py
+++ b/apostrophe/export_dialog.py
@@ -69,7 +69,7 @@ class Format(GObject.Object):
 
     @property
     def requires_texlive(self):
-        return self.ext in {"tex", "pdf"} and self.engine != "typst"
+        return self.ext in {"tex", "pdf"} and self.engine != "weasyprint"
 
 
 class ExportDialog:
@@ -83,9 +83,8 @@ class ExportDialog:
             "extension": "pdf",
             "to": "pdf",
             "mimetype": "application/pdf",
-            "args": ["--pdf-engine=typst",
-                     "--variable=papersize:a4",
-                     "-V", "template=%s" % helpers.get_media_path('/pandoc_templates/typst.typ')]
+            "args": ["--pdf-engine=weasyprint",
+                     "--variable=papersize:a4"]
         },
         "html":
         {
@@ -243,7 +242,7 @@ class AdvancedExportDialog(Adw.Dialog):
 
     @GObject.Property(type=bool, default=False)
     def show_syntax_options(self):
-        return self.formats_list.get_selected_row().item.has_syntax and self.formats_list.get_selected_row().item.engine != "typst"
+        return self.formats_list.get_selected_row().item.has_syntax
 
     @GObject.Property(type=bool, default=False)
     def show_presentation_options(self):
@@ -346,8 +345,6 @@ class AdvancedExportDialog(Adw.Dialog):
 
         if self.formats_list.get_selected_row().item.ext == "pdf":
             args.append(f"--pdf-engine={self.formats_list.get_selected_row().item.engine}")
-            if self.formats_list.get_selected_row().item.engine == "typst":
-                args.extend(['-V', 'template=%s' % helpers.get_media_path('/pandoc_templates/typst.typ')])
 
         if self.sw_standalone.get_active():
             args.append("--standalone")
diff --git a/apostrophe/main_window.py b/apostrophe/main_window.py
index 7d73aac..6d43bcf 100644
--- a/apostrophe/main_window.py
+++ b/apostrophe/main_window.py
@@ -88,8 +88,6 @@ class MainWindow(Adw.ApplicationWindow):
         super().__init__(application=Gio.Application.get_default(),
                          title="Apostrophe")
 
-        os.chdir("/")
-
         # Preferences
         self.settings = Settings.new()
 
diff --git a/data/media/formats.json b/data/media/formats.json
index 07cb846..cc4643f 100644
--- a/data/media/formats.json
+++ b/data/media/formats.json
@@ -3,7 +3,7 @@
       "name": "PDF",
       "ext": "pdf",
       "to": "pdf",
-      "engine": "typst"
+      "engine": "weasyprint"
     },
     {
       "name": "PDF (LaTeX)",
-- 
GitLab

